Source/FlowEditor/Private/DetailCustomizations/FlowNode_CustomNodeBaseDetails.cpp

```diff
@@ -0,0 +1,152 @@
+// Copyright https://github.com/MothCocoon/FlowGraph/graphs/contributors
+
+#include "DetailCustomizations/FlowNode_CustomNodeBaseDetails.h"
+#include "FlowAsset.h"
+#include "Nodes/Route/FlowNode_CustomNodeBase.h"
+
+#include "DetailCategoryBuilder.h"
+#include "DetailWidgetRow.h"
+#include "PropertyEditing.h"
+#include "Widgets/Input/SComboBox.h"
+#include "Widgets/Text/STextBlock.h"
+#include "Widgets/SWidget.h"
+
+
+void FFlowNode_CustomNodeBaseDetails::CustomizeDetails(IDetailLayoutBuilder& DetailLayout)
+{
+	// Subclasses must override this function (and call CustomizeDetailsInternal with the localized text)
+	checkNoEntry();
+}
+
+void FFlowNode_CustomNodeBaseDetails::CustomizeDetailsInternal(IDetailLayoutBuilder& DetailLayout, const FText& CustomRowNameText, const FText& EventNameText)
+{
+	DetailLayout.GetObjectsBeingCustomized(ObjectsBeingEdited);
+
+	if (ObjectsBeingEdited[0].IsValid())
+	{
+		UFlowNode_CustomNodeBase* FlowNodeBase = CastChecked<UFlowNode_CustomNodeBase>(ObjectsBeingEdited[0]);
+		CachedEventNameSelected = MakeShared<FName>(FlowNodeBase->GetEventName());
+	}
+
+	IDetailCategoryBuilder& Category = CreateDetailCategory(DetailLayout);
+
+	Category.AddCustomRow(CustomRowNameText)
+		.NameContent()
+			[
+				SNew(STextBlock)
+					.Text(EventNameText)
+			]
+		.ValueContent()
+			.HAlign(HAlign_Fill)
+			[
+				SAssignNew(EventTextListWidget, SComboBox<TSharedPtr<FName>>)
```

Comment:
_Storing a pointer to the widget so that we can refresh it later in **PinSelectionChanged**_